python/cuspatial/cuspatial/core/geodataframe.py

```diff
@@ -202,6 +202,73 @@ def _gather(
         # return
         return result
 
+    def reset_index(
+        self, level=None, drop=False, inplace=False, col_level=0, col_fill=""
+    ):
+        """Reset the index, or a level of it.
+
+        Parameters
+        ----------
+        level : `int`, `str`, `tuple`, or `list`, default `None`
+            Only remove the given levels from the index. Removes all levels by
+            default.
+        drop : `bool`, default `False`
+            Do not try to insert index into dataframe columns. This resets the
+            index to the default integer index.
+        inplace : `bool`, default `False`
+            Modify the GeoDataFrame in place (do not create a new object).
+        col_level : `int` or `str`, default `0`
+            If the columns have multiple levels, determines which level the
+            labels are inserted into. By default it is inserted into the first
+            level.
+        col_fill : `object`, default `""`
+            If the columns have multiple levels, determines how the other
+            levels are named. If None then the index name is repeated.
+
+        Returns
+        -------
+        `GeoDataFrame`
+        """
+
+        # Split geometry and non-geometry columns
+        geo_data, cudf_data = self._split_out_geometry_columns()
+
+        # Reset cudf column
+        reindexed = cudf_data.reset_index(
+            level, drop, inplace, col_level, col_fill
+        )
+
+        # Reset GeoColumns
+        recombiner = self.copy(deep=False)
+        recombiner.index = cudf.RangeIndex(len(recombiner))
+        if reindexed:
+            # Not a multi-index, and the index was not dropped.
+            if "index" in reindexed.columns:
+                recombiner.insert(
+                    loc=0, name="index", value=reindexed["index"]
+                )
+            # If the index is a MultiIndex, we need to insert the
+            # individual levels into the GeoDataFrame.
+            elif "level" in reindexed.columns[0]:
```

Comment:
Similarly: (You can even factor `if not drop`)
```suggestion
            elif not drop and isinstance(cudf_data.index, cudf.MultiIndex):
```

Reply:
Done with both.